driver-sync/src/test/functional/com/mongodb/client/unified/Entities.java

```diff
@@ -529,30 +522,42 @@ private void initClient(final BsonDocument entity, final String id,
                                 ServerMonitoringModeUtil.fromString(value.asString().getValue())));
                         break;
                     case "authMechanism":
-                        if (value.equals(new BsonString(AuthenticationMechanism.MONGODB_OIDC.getMechanismName()))) {
-                            clientSettingsBuilder.credential(MongoCredential.createOidcCredential(null));
+                        if (value.equals(new BsonString(MONGODB_OIDC.getMechanismName()))) {
+                            // authMechanismProperties depends on authMechanism
+                            BsonDocument authMechanismProperties = entity
+                                    .getDocument("uriOptions")
+                                    .getDocument("authMechanismProperties");
+                            boolean hasPlaceholder = authMechanismProperties.equals(
+                                    new BsonDocument("$$placeholder", new BsonInt32(1)));
+                            if (!hasPlaceholder) {
+                                throw new UnsupportedOperationException("Unsupported authMechanism: " + value);
```

Comment:
It's not the `authMechanism` that is not supported here, but any `authMechanismProperties` that is not equal to `{ "$$placeholder": 1 }`. The exception message, however, says `Unsupported authMechanism` and includes `value` (the authentication mechanism) instead of including `authMechanismProperties`.